engines/asylum/puzzles/puzzle11.cpp

```diff
@@ -0,0 +1,129 @@
+
+#include "asylum/views/scene.h"
+
+#include "asylum/asylum.h"
+
+
+namespace Asylum {
+
+Puzzle11::Puzzle11(AsylumEngine *engine) : Puzzle(engine) {
+	_counter = 0;
+	memset(&_frameCounts, 0, sizeof(_frameCounts));
+	memset(&_frameIndexes, 0, sizeof(_frameIndexes));
+	_data_457298 = false;
+	_data_45AA34 = false;
+}
+
+Puzzle11::~Puzzle11() {
+}
+
+void Puzzle11::saveLoadWithSerializer(Common::Serializer &s) {
+	s.syncAsUint32LE(_counter);
+
+	s.skip(8); // 2 int32 that do not seem to be used
+
+	s.syncAsUint32LE(_frameIndexes[0]);
+	s.syncAsUint32LE(_frameIndexes[3]);
+
+	warning("[Puzzle11::saveLoadWithSerializer] Not implemented");
+	s.skip(8);
+	s.syncAsUint32LE(_frameIndexes[2]);
+	s.skip(4);
+	s.syncAsUint32LE(_frameIndexes[1]);
+	s.syncAsUint32LE(_frameIndexes[4]);
+	s.skip(28);
+	s.syncAsUint32LE(_frameIndexes[5]);
+	s.syncAsUint32LE(_frameIndexes[6]);
+}
+
+//////////////////////////////////////////////////////////////////////////
+// Event Handling
+//////////////////////////////////////////////////////////////////////////
+bool Puzzle11::init(const AsylumEvent &evt)  {
+	// FIXME! getPuzzleData()->timeMachineCounter = 0;
+
+	getScreen()->setPalette(getWorld()->graphicResourceIds[12]);
+	getScreen()->setGammaLevel(getWorld()->graphicResourceIds[12]);
+	getCursor()->set(getWorld()->graphicResourceIds[60], -1, kCursorAnimationNone, 7);
+
+	_data_45AA34 = false;
+
+	_frameCounts[0] = GraphicResource::getFrameCount(_vm, getWorld()->graphicResourceIds[59]);
+	_frameCounts[1] = GraphicResource::getFrameCount(_vm, getWorld()->graphicResourceIds[54]);
+	_frameCounts[2] = GraphicResource::getFrameCount(_vm, getWorld()->graphicResourceIds[55]);
+	_frameCounts[3] = GraphicResource::getFrameCount(_vm, getWorld()->graphicResourceIds[56]);
+	_frameCounts[4] = GraphicResource::getFrameCount(_vm, getWorld()->graphicResourceIds[13]);
+
+	mouseLeftDown(evt);
+
+	return true;
+}
+
+bool Puzzle11::update(const AsylumEvent &)  {
+	getScreen()->clearGraphicsInQueue();
+	getScreen()->draw(getWorld()->graphicResourceIds[11]);
+	getScreen()->addGraphicToQueue(getWorld()->graphicResourceIds[15], _frameIndexes[0], Common::Point(122, 269), kDrawFlagNone, 0, 1);
+	getScreen()->addGraphicToQueue(getWorld()->graphicResourceIds[16], _frameIndexes[1], Common::Point(524, 175), kDrawFlagNone, 0, 1);
+	getScreen()->addGraphicToQueue(getWorld()->graphicResourceIds[18], _frameIndexes[2], Common::Point(359,  61), kDrawFlagNone, 0, 1);
+	getScreen()->addGraphicToQueue(getWorld()->graphicResourceIds[14], _frameIndexes[3], Common::Point(115,  80), kDrawFlagNone, 0, 1);
+	getScreen()->addGraphicToQueue(getWorld()->graphicResourceIds[17], _frameIndexes[4], Common::Point(238,  94), kDrawFlagNone, 0, 1);
+
+	getScreen()->drawGraphicsInQueue();
+	getScreen()->copyBackBufferToScreen();
+	warning("[Puzzle11::update] Not implemented!");
+
+	return true;
+}
+
+bool Puzzle11::mouseRightDown(const AsylumEvent &) {
+	warning("[Puzzle11::mouseUp] Not implemented!");
```

Comment:
Does it mean that this puzzle is not yet working?

Reply:
This puzzle doesn't seem to be used in the original game. I couldn't find any resources for it